src/components/ng2-map.component.ts

```diff
@@ -114,7 +114,7 @@ export class Ng2MapComponent implements OnChanges, OnDestroy, AfterViewInit {
       window['ng2MapRef'].forEach( ng2MapRef => {
         ng2MapRef.zone.run(function() { ng2MapRef.componentFn(); });
       });
-      window['ng2MapRef'] = [];
+      window['ng2MapRef'].splice(0, window['ng2MapRef'].length);
```

Comment:
Sorry, we should see function initializeMap() first. At the last of this function, the map object is setted to window.ng2MapRef. 

`window['ng2MapRef'].map = this.map;`

So window.ng2MapRef is not simply an array at all, it contains the map object, Although it is commended for useages of test and debug, but the users will need the map object anywhere outside the range of map.
For example, maybe there is a button outside the map, and I want to call map.fitBounds to show a polygon or all of many markers on the map...

Then, we see `window['ng2MapRef'] = [];` at the function addGoogleMapsApi(), it will clear the array and also the map object.

Also, I'm thinking of moving those ng2MapRef and initNg2Map to some services such as ng2Map.

At the last, may i got some misunderstand of the ng2Map? 
If there is, please teach me as well.